subsys/net/lib/dns/mdns_responder.c

```diff
@@ -333,6 +333,11 @@ static void send_sd_response(struct net_context *ctx,
 					   &ip_hdr->ipv6->src);
 	}
 
+	/*
+	 * XXX: need to modify the function below to optionally extract instance,
+	 * if it exists in the payload. Probably will want to unit test the heck
+	 * out of it afterward.
+	 */
```

Comment:
`dns_sd_extract_service_proto_domain()` will need to be modified to also *optionally* extract instance.

* pass in an `instance_buf`
* if non-`NULL`, and size is OK, then extract instance
* in the end, will need to fill 3, maybe 4 buffers, so a bit of a rework will be required.

Reply:
Filed issue #38846